### PR TITLE
[Java] Generated API class has wrong dataType and does not compile Issue

### PR DESCRIPTION
**Pull request by sebastien-rosset**
Details of the issue are described in #5324.

Without this PR, the generated code is as below. There are two problems: 1) the type of the requestBody argument is wrong (it should be SyslogPolicy, not PolicyAbstractPolicy); 2) The type of the requestBody argument uses generics.
 
```
createSyslogPolicyCall(PolicyAbstractPolicy<String, Object> requestBody ....
```

With this PR, the generated code is as follows, it fixes both problems:
```
createSyslogPolicyCall(SyslogPolicy requestBody ....
```

This is not ready to be merged, but I need a bit of guidance to make progress.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [ ] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 1]
```diff
@@ -1752,6 +1752,10 @@ public String getSchemaType(Schema schema) {
      */
     @SuppressWarnings("static-method")
     public String toAllOfName(List<String> names, ComposedSchema composedSchema) {
+        Map<String, Object> exts = composedSchema.getExtensions();
+        if (exts != null && exts.containsKey("x-allOf-name")) {
+            return (String) exts.get("x-allOf-name");
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java]: We want to standardize vendor extension naming with lower cases only separated by hyphens so  what about using`x-all-of-name` instead?

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team  as the change covers default codegen.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaModelEnumTest.java [hunk 2]
```diff
@@ -172,6 +172,7 @@ public void testEnumTestSchema() {
         codegen.setOpenAPI(openAPI);
 
         Schema enumTest = openAPI.getComponents().getSchemas().get("Enum_Test");
+        Assert.assertNotNull(enumTest);
```
Comment (spacether) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/java/JavaModelEnumTest.java]: Do our tests cover this new code?
This model: https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml#L1422
does not contain `x-all-of-name`
Also doesn't this test require that the schema not be null?
Down below we test that the CodegenModel generated form the schema contains vars, which means that the schema had to not be null to contain those vars.

**Comment by sebastien-rosset** [comment]
> Do our tests cover this new code?

The reason I added this line is only to make the test more robust, it's not really related to this PR. At some point during my testing, I mistakenly changed the content of the OAS document which is used for this enum test. It ended up returning a null value for "enumTest", and it caused a null pointer exception at the next line. So even if enumTest is null, the next line will fail, which is what is expected, but it's better to assert that enumTest is not null rather than relying on a NPE to be raised.

> Also doesn't this test already require that the schema not be null?
Yes, but this above, this makes the UT code more robust.

**Comment by sebastien-rosset** [comment]
The "avro-schema" fake-petstore generator fails in the CI process. I found out the .drone file invokes bin/utils/test-fake-petstore-for-all.sh, which in turns executes a java command that redirects all output to /dev/null, so it's not possible to troubleshoot by directly looking at the CI logs. I then executed the script locally, but the execution completes successfully in my local workspace. There must be something different in the CI execution environment compared to my local workspace.
To get around the problem, I have changed the bin/utils/test-fake-petstore-for-all.sh script to redirect the output to a file, and if the execution fails, cat the log file.

```
ERROR: Failed to run avro-schema
--
140 | java -jar ./modules/openapi-generator-cli/target/openapi-generator-cli.jar generate -i modules/openapi-generator/src/test/resources/2_0/petstore-with-fake-endpoints-models-for-testing.yaml -g avro-schema -o /tmp/openapi-generator-test-fake-petstore/2.0/avro-schema
```

**Comment by wing328** [comment]
Looks good but I've not tested it locally
